src/streams/register_nack.rs

```diff
@@ -196,6 +245,35 @@ impl NackRegister {
         )
     }
 
+    /// Returns the next time a NACK needs to be sent, if any
+    pub fn next_nack_time(&self, now: Instant, rtt: Duration) -> Option<Instant> {
+        let Range { start, end } = self.active.as_ref()?;
+
+        let mut earliest: Option<Instant> = None;
+
+        // Iterate over u64 values, then convert back to SeqNo
+        for s_u64 in **start..=**end {
+            let packet = self.packet(s_u64.into());
+            if packet.received || packet.nack_count >= MAX_NACKS {
+                continue;
+            }
+
+            let need_at = if let Some(sent_at) = packet.sent_at {
+                sent_at + rtt
```

Comment:
Maybe we should add a little more time here. I mean, is RTT considered as the largest possible round trip time or is it just the last round trip time. If it's the last round trip time, then it would make sense to multiply it by 1.2, for example